force-app/main/SDO_SCOM_TNC_ACKNOWLEDGEMENT/lwc/sdo_scom_tnc_acknowledgement_place_order/sdo_scom_tnc_acknowledgement_place_order.js

```diff
@@ -1,43 +1,53 @@
-import { api } from 'lwc';
-import { CheckoutComponentBase } from 'commerce/checkoutApi';
+import { LightningElement, api } from 'lwc';
+import { useCheckoutComponent } from "commerce/checkoutApi";
 
-export default class Sdo_scom_tnc_acknowledgement_place_order extends CheckoutComponentBase {
-    _checkedByDefault = false;
+export default class Sdo_scom_tnc_acknowledgement_place_order extends useCheckoutComponent(LightningElement) {
     checked = false;
     showError = false;
-    checkoutDisabled = true;
 
     @api error = 'Please click the checkbox to accept the terms and conditions';
 
     @api termsAndConditions = "Sit amet mattis vulputate enim nulla aliquet porttitor lacus. Ut tristique et egestas quis ipsum suspendisse ultrices gravida. Ut porttitor leo a diam. Magna fermentum iaculis eu non. Sit amet justo donec enim. Quis eleifend quam adipiscing vitae. Amet nisl purus in mollis nunc sed id semper. Leo vel fringilla est ullamcorper. Quis vel eros donec ac odio tempor orci dapibus. Curabitur vitae nunc sed velit dignissim sodales ut. Maecenas sed enim ut sem viverra aliquet eget sit amet. Felis donec et odio pellentesque diam. Ultricies mi quis hendrerit dolor magna eget. Eu facilisis sed odio morbi. Aliquet enim tortor at auctor urna nunc id cursus metus.";
 
-    @api checkoutDetails;
+    @api termsAndConditionsHeading = 'Terms & Conditions';
 
-    @api buttonTitle = 'Place Order';
+    stageAction(checkoutStage) {
+        if (checkoutStage === 'CHECK_VALIDITY_UPDATE') {
```

Comment:
In our case its not needed as we are not updating anything. I think save check is enough.